### PR TITLE
Fix memory stats printing

### PR DESCRIPTION
**Pull request by pengwa**
### Fix memory stats printing

The mmeory stats printing is failed when module is in eval mode, doing ORTModule wrap. At that time, runtime inspector for training manager should have training model being true, but got a false (because existing logic get the boolean from module.training). Runtime inspector as part of training manager or inference manager should know it is serving training or not explicitly, so we cannot depend on the stat of module.training during ORTModule initialization.

### CODE REVIEW

**Review by wschin — APPROVED** [review]

File: orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py [hunk 1]
```diff
@@ -6496,3 +6496,39 @@ def run_step(model, x, y, z):
     torch.cuda.synchronize()
     if original_val is not None:
         os.environ["ORTMODULE_MEMORY_OPT_LEVEL"] = original_val
+
+
+def test_bert_memory_inspection(caplog):
+    original_val = os.environ.get("ORTMODULE_PRINT_MEMORY_STATS", None)
```
Comment (wschin) [inline orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py:6502]: Not sure if this env var is really set during testing.

Comment (pengwa) [reply orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py:6502]: yeah, nobody actually set it unless we want to enable that locally when running tests. This is a common approach to handle the env var in the test, so this is to follow that. :)

File: orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py [hunk 2]
```diff
@@ -54,6 +54,7 @@ def __init__(
         self,
         module: _FlattenedModule,
         debug_options: DebugOptions,
+        export_mode: int,
```
Comment (wschin) [inline orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py:57]: Add doc string to explain this.

File: orttraining/orttraining/python/training/ortmodule/_runtime_inspector.py [hunk 3]
```diff
@@ -479,7 +479,7 @@ class MemoryObserver:
     NORMALIZER_FACTOR = float(1024 * 1024)
     NORMALIZER_UNIT = "MiB"
 
-    def __init__(self, m: torch.nn.Module, logger: Logger):
+    def __init__(self, m: torch.nn.Module, logger: Logger, training: bool):
```
Comment (wschin) [inline orttraining/orttraining/python/training/ortmodule/_runtime_inspector.py:482]: doc string for new parameter.

Comment (pengwa) [reply orttraining/orttraining/python/training/ortmodule/_runtime_inspector.py:489]: added

File: orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py (hunk 2 above)
Comment (pengwa) [reply orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py:57]: added

File: orttraining/orttraining/python/training/ortmodule/_inference_manager.py [hunk 4]
```diff
@@ -28,8 +28,7 @@ class InferenceManager(GraphExecutionManager):
     """
 
     def __init__(self, model, debug_options: DebugOptions, fallback_manager: _FallbackManager, logger: Logger):
-        super().__init__(model, debug_options, fallback_manager, logger)
-        self._export_mode = torch.onnx.TrainingMode.EVAL
+        super().__init__(model, debug_options, torch.onnx.TrainingMode.EVAL, fallback_manager, logger)
```
Comment (guyang3532) [inline orttraining/orttraining/python/training/ortmodule/_inference_manager.py:31]: Need the TrainingManager add this arg in constructor?

**Review by guyang3532 — APPROVED** [review]
